### ISSUE TITLE
StreamRpcChannel uses root logger

### ISSUE THREAD

**Issue by shenghaoyang** [OpenLightingProject/ola#1412]
Hello, 

I've been trying to use the python library to access OLA without using the wrapper class (so I can integrate the client into my own event loop). 

While going over the source code to figure out how the client works, I've found multiple logging calls that send log messages straight to the root logger in  ``ola/python/ola/rpc/StreamRpcChannel.py``.

Example: 
```
  def SocketReady(self):
    """Read data from the socket and handle when we get a full message.
    Returns:
      True if the socket remains connected, False if it was closed.
    """
    data = self._socket.recv(self.RECEIVE_BUFFER_SIZE)
    if data == '':
      logging.info('OLAD Server Socket closed')
      if self._close_callback is not None:
```

Is this intentional? There's no _clean_ way to prevent the OLA library from polluting the logs with its own messages, because OLA does not source a logger for itself. I could source my own logger for the main application and change the handler for the root logger to the null handler, but I don't think OLA should be directly logging to the root logger in the first place.

Regards, 

Shenghao Yang

**Comment by peternewman** [comment]
Hi Shenghao,

I suspect not. I assume you mean we should inherit from our own logger, so the logging level for OLA can be set to a different level, or redirected to a file/STDERR etc?

Pull requests welcome anyway, if you'd like to fix it up to how we should be doing it. Should our Python CLI tools be changed too in python/examples? And the RDM tests in tools/rdm?

In terms of integrating with your own event loop, have you seen the info here:
https://docs.openlighting.org/ola/doc/latest/event_driven.html#event_Integration

**Comment by shenghaoyang** [comment]
Hi, 

> I suspect not. I assume you mean we should inherit from our own logger, so the logging level for OLA can be set to a different level, or redirected to a file/STDERR etc?

Yeah, I think the library should obtain its own logger and log there instead, so the user of the library can choose whether log messages from OLA are shown, attach them to handlers they want, and mask them out instead of messing around with the log hierarchy, or resorting to filtering.

> Should our Python CLI tools be changed too in python/examples? And the RDM tests in tools/rdm?

Since it seems the tools are meant to be executed directly, there shouldn't be a problem with use of the root logger, since it doesn't interfere with any other output (if they were meant to be output, anyway). Unless someone needs to integrate those tools into their own projects, it wouldn't really be a problem.

> Pull requests welcome anyway, if you'd like to fix it up to how we should be doing it.

I'm probably not the best choice for this, but I'll see if I can alter things a bit. Honestly it isn't much of a problem, just the differing output in the logs making me a little crazy. 

> In terms of integrating with your own event loop, have you seen the info here:
> https://docs.openlighting.org/ola/doc/latest/event_driven.html#event_Integration

Thanks for that. Didn't realize it was there the first time I looked through it 
(error between keyboard and chair), but it does help to confirm that I'm doing things 
in sort-of "the right way". 

Best, 

Shenghao

**Comment by shenghaoyang** [comment]
Resolved in #1415
